### PR TITLE
feat: window_max 

### PR DESCRIPTION
**Pull request by xabi**
limited impl of window_max for 4d tensors using max_pool2d on the candle-core side

### CODE REVIEW

File: lib/candlex/native.ex [hunk 1]
```diff
@@ -55,6 +55,7 @@ defmodule Candlex.Native do
   def pad_with_zeros(_tensor, _left, _right), do: error()
   def clamp(_tensor, _min, _max), do: error()
   def reverse(_tensor, _axes), do: error()
+  def max_pool2d(_tensor, _dimensions, _strides), do: error()
```
Comment (grzuy) [inline lib/candlex/native.ex:58]: Not that there's anything wrong with this, just to be consistent with the rest of the file...
```suggestion
  def max_pool2d(_tensor, _dims, _strides), do: error()
```

File: native/candlex/src/tensors.rs [hunk 2]
```diff
@@ -390,6 +390,36 @@ pub fn conv2d(
     )?))
 }
 
+#[rustler::nif(schedule = "DirtyCpu")]
+pub fn max_pool2d(
+    tensor: ExTensor,
+    dimensions: Vec<usize>,
```
Comment (grzuy) [inline native/candlex/src/tensors.rs:396]: Same here about argument name `dims`

**Review by grzuy — APPROVED** [review]
Neat! :100: 

Thank you very much!

File: lib/candlex/backend.ex [hunk 3]
```diff
@@ -488,6 +488,28 @@ defmodule Candlex.Backend do
     |> to_nx(out)
   end
 
+  @impl true
+  def window_max(%T{type: out_type} = out, tensor, window_dimensions, opts) do
+    win_dims =
+      case window_dimensions do
+        {1, 1, dx, dy} -> [dx, dy]
+        dims -> raise("unsupported window dimensions #{inspect(dims)}")
+      end
```
[the remark this answers is not in the document]
Comment (xabi) [reply lib/candlex/backend.ex:497]: sure !

File: native/candlex/src/tensors.rs (hunk 2 above)
Comment (xabi) [reply native/candlex/src/tensors.rs:396]: okay i'll update the this PR and create an other PR for window_mean in the same vein tomorrow. Thx for your comments !

**Comment by grzuy** [comment]
@xabi Thank you for the updates :raised_hands: 

FYI, I added an additional commit simplifying a bit the NIF so that it directly receives tuples instead of vecs.

**Comment by grzuy** [comment]
Published `v0.1.6`.

File: native/candlex/src/tensors.rs (hunk 2 above)
Comment (grzuy) [reply native/candlex/src/tensors.rs]: > okay i'll update this PR and create an other PR for window_mean in the same vein tomorrow. Thx for your comments !

Just realized `Nx.window_mean` doesn't need a `window_mean` implemented in the backend. It uses the backend `window_sum`, so we actually need `Candlex.Backend.window_sum` to support both `Nx.window_sum` and `Nx.window_mean`.

**Pull request merged**